app/controllers/application_controller.rb

```diff
@@ -1,4 +1,35 @@
 class ApplicationController < ActionController::Base
-  # Only allow modern browsers supporting webp images, web push, badges, import maps, CSS nesting, and CSS :has.
+  before_action :configure_permitted_parameters, if: :devise_controller?
   allow_browser versions: :modern
+
+  protected
+
+  def configure_permitted_parameters
+    devise_parameter_sanitizer.permit(:sign_up, keys: %i[name])
+    devise_parameter_sanitizer.permit(:account_update, keys: %i[name])
+  end
+
+  def current_cart
+    if user_signed_in?
+      find_or_create_user_cart
+    else
+      find_or_create_guest_cart
+    end
+  end
+
+  def find_or_create_user_cart
+    cart = current_user.cart || current_user.create_cart!
+    cart.merge_guest_cart(session.delete(:cart_id))
+    cart
+  end
+
+  def find_or_create_guest_cart
+    if session[:cart_id]
+      Cart.find(session[:cart_id])
+    else
+      cart = Cart.create!
+      session[:cart_id] = cart.id
+      cart
+    end
```

Comment:
`session[:cart_id]` は既存カートがあってもなくてもセットしてよいとおもうので、これでもいけるかも。
```suggestion
    cart = Cart.find_or_create_by!(session[:cart_id])
    session[:cart_id] = cart.id
```